provider/azure/internal/imageutils/images_test.go

```diff
@@ -47,13 +47,13 @@ func (s *imageutilsSuite) SetUpTest(c *gc.C) {
 
 func (s *imageutilsSuite) TestBaseImageOldStyle(c *gc.C) {
 	s.mockSender.AppendResponse(azuretesting.NewResponseWithContent(
-		`[{"name": "20_04-gen2"}, {"name": "20_04-lts"}, {"name": "20_04-arm64"}]`,
+		`[{"name": "20_04-lts-gen2"}, {"name": "20_04-lts"}, {"name": "20_04-lts-arm64"}]`,
 	))
-	image, err := imageutils.BaseImage(s.callCtx, corebase.MakeDefaultBase("ubuntu", "20.04"), "released", "westus", s.client)
+	image, err := imageutils.BaseImage(s.callCtx, corebase.MakeDefaultBase("ubuntu", "20.04"), "released", "westus", s.client, false)
```

Comment:
All of these tests call BaseImage with `false`. 
Are there tests for the gen 2 images?
We'd need gen2 legacy and new

Reply:
Added more tests.